File: engine/src/main/java/com/bartlomiejpluta/base/engine/world/tileset/asset/TileSetAsset.java
package com.bartlomiejpluta.base.engine.world.tileset.asset;

import com.bartlomiejpluta.base.engine.common.asset.Asset;
import lombok.Getter;
import lombok.NonNull;

@Getter
public class TileSetAsset extends Asset {
   private final int rows;
   private final int columns;

   public TileSetAsset(@NonNull String uid, @NonNull String source, int rows, int columns) {
      super(uid, source);
      this.rows = rows;
      this.columns = columns;
   }
}

File: editor/src/main/kotlin/com/bartlomiejpluta/base/editor/tileset/asset/TileSetAsset.kt
package com.bartlomiejpluta.base.editor.tileset.asset

import com.bartlomiejpluta.base.editor.asset.model.GraphicAsset
import com.bartlomiejpluta.base.editor.project.model.Project

class TileSetAsset(project: Project, uid: String, source: String, name: String, rows: Int, columns: Int) :
   GraphicAsset(project.tileSetsDirectoryProperty, uid, source, name, rows, columns)
